### DISCUSSION TITLE
Lossless WEBP

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7700] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

Can ImageMagick make lossless webp files? If so, how?

**Comment by snibgo** [comment]
```
magick in.png -quality 100 -define webp:lossless=true out.webp
```
If in.png is 8 bits/channel/pixel, out.webp will be identical to the input (I think). If the input is 16 bits/channel/pixel, x.webp will be almost identical to the input.

See https://imagemagick.org/script/webp.php for other webp options.

EDIT: I should mention that webp records only 8 bits per channel, so images that need more precision than this will lose precision when stored as webp. In my view, webp is a useful final format when no more image processing is needed, but should not be used as an intermediate format.
